sql/importer: TestImportDefaultNextVal failed

sql/importer.TestImportDefaultNextVal [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12218755?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12218755?buildTab=artifacts#/) on release-23.2 @ [f151df1bc3d16a2b307852a7897e8c0f78050ebc](https://github.com/cockroachdb/cockroach/commits/f151df1bc3d16a2b307852a7897e8c0f78050ebc):


```
=== RUN   TestImportDefaultNextVal
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestImportDefaultNextVal3785521019
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    import_stmt_test.go:4762: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestImportDefaultNextVal3785521019
--- FAIL: TestImportDefaultNextVal (50.32s)
=== RUN   TestImportDefaultNextVal/nextval
    --- FAIL: TestImportDefaultNextVal/nextval (19.23s)
=== RUN   TestImportDefaultNextVal/nextval/two-nextval-diff-seq
    import_stmt_test.go:4757: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/importer/import_stmt_test.go:4757
        	Error:      	Not equal: 
        	            	expected: 1116
        	            	actual  : 1119
        	Test:       	TestImportDefaultNextVal/nextval/two-nextval-diff-seq
        --- FAIL: TestImportDefaultNextVal/nextval/two-nextval-diff-seq (5.67s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #110788 sql/importer: TestImportDefaultNextVal failed [C-test-failure O-robot T-sql-queries branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportDefaultNextVal.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32490

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -4649,7 +4649,9 @@ func TestImportDefaultNextVal(t *testing.T) {
 					if util.RaceEnabled {
 						expectedVal = test.seqToNumNextval[seqName].expectedImportChunkAllocsUnderRace
 					}
-					require.Equal(t, expectedVal, seqVal)
+					// The seqVal may have advanced further due to retries, so it may be
+					// greater than the expectedVal.
+					require.LessOrEqual(t, expectedVal, seqVal)
 				}
 			})
 		}

